tests/cloudpickle_test.py

```diff
@@ -2116,19 +2128,38 @@ def check_annotations(obj, expected_type):
                 assert check_annotations(obj, type_) == "ok"
                 assert worker.run(check_annotations, obj, type_) == "ok"
 
-    def test_generic_extensions(self):
+    def test_generic_extensions_literal(self):
         typing_extensions = pytest.importorskip('typing_extensions')
 
-        objs = [
-            typing_extensions.Literal,
-            typing_extensions.Final,
-            typing_extensions.Literal['a'],
-            typing_extensions.Final[int],
+        def check_literal_equal(obj1, obj2):
+            assert obj1.__values__ == obj2.__values__
+            assert type(obj1) == type(obj2) == typing_extensions._LiteralMeta
+        literal_objs = [
+            typing_extensions.Literal, typing_extensions.Literal['a']
         ]
+        for obj in literal_objs:
+            depickled_obj = pickle_depickle(obj, protocol=self.protocol)
+            try:
+                # __eq__ does not work for Literal objects in early Python 3.5
```

Comment:
I think this test should inspect the Python version to know which assertions to check instead of using a try / except block.

Reply:
Sure.